### DISCUSSION TITLE
Edge attributes in Locate API response

### DISCUSSION THREAD

**Discussion by Bondify** [valhalla/valhalla#4351] [Q&A] [answered]
hey there, I'm using the Locate API to get snap some points and get attributes of the street edges that were included in the response. I noticed that most times I get a couple of edges that have different ids but the exact same geometry. Digging into it I noticed they have different values for the label `forward` (one true and the other one false). 
Are these two edges actually the same in different directions? Would this be equivalent to the `reverse` tag from OSM.

Thanks a lot for your help and for maintaining Valhalla, it's an amazing tool.

**Comment by nilsnolde** [comment, marked as answer]
OSM != Graph. In our graph, every way segment is represented by 2 directed edges, even it’s just one way in OSM. And even if it’s a one-way, then the other direction is just not accessible. For a users sake it’s an implementation detail. But yes, one will be in forward direction (whatever direction OSM has it in), the other in reverse.

**Reply by Bondify** [reply]
amazing, thanks a lot for your fast response. This is exactly what I needed to know.

**Reply by Bondify** [reply]
@nilsnolde one more question: when you say one direction might not be accessible, is that information under the `not-thru` tag? Thanks again

**Reply by kevinkreiser** [reply]
no each edge has a section called `access` and in there each mode of travel gets a boolean indicating whether it can or cannot use this edge in this direction
